song-client/src/main/java/org/icgc/dcc/song/client/register/Registry.java

```diff
@@ -128,7 +138,61 @@ public Status search(String studyId,
       String donorId,
       String fileId){
     val url = endpoint.searchGet(studyId,sampleId,specimenId,donorId,fileId);
-    return restClient.get(accessToken, url);
+    return tryGet(accessToken, url);
+  }
+
+  /**
+   * Wraps the restClient calls, so that unauthorized.token errors are handled properly. In addition, if an error occurs, the availability of the song server is checked, to ensure its running.
+   * @param supplier Request method that returns a Status object
+   * @return the status
+   */
+  private Status tryClient(Supplier<Status> supplier){
```

Comment:
Why write specific error handling for just the REST client exceptions?

Put one try/catch block around the run method in ClientMethod, and you can catch and report on all the
various exception conditions in one place. That's simpler than passing around lambda functions, in my opinion.